app/models/host/managed.rb

```diff
@@ -926,6 +927,17 @@ def apply_compute_profile(modification)
 
   private
 
+  def owner_taxonomies_match
+    if self.owner.is_a?(User)
+      if Organization.organizations_enabled && self.organization_id && !self.owner.my_organizations.where(:id => self.organization_id).exists?
```

Comment:
> self.owner.organization_and_child_ids.include?(self.organization_id) so you can avoid adding new methods in user.rb, location.rb and organization.rb

`User#my_organizations` uses `Organization.my_organizations` scope which reflects admin flag. If admin was not assigned to any org/loc (usually the case) this would always return false. `organization_and_child_ids` should only be used "internally" e.g. in `Organization.my_organizations` scope. Otherwise we'd had to add admin condition everywhere.

> Why not implementing these through Taxonomy.enabled_taxonomies by the way? It's basically duplicated code as is now

Since you asked [second time](https://github.com/theforeman/foreman/pull/3704#discussion_r74396509), I'll try to explain a bit more why I want to avoid it. I don't think this is duplication because
* Organization and Location are two different things. The implementation makes it easy to use them in same way leading to problems like when you later realize that Location should not act as Organization you'd have to split the logic everywhere. For example the taxonomy inheritance does not always make sense for both, should Subnet be shared with all sub-orgs? I'd say no. Should Subnet be shared with sub-locations? That makes more sense to me. I think it was mistake to use STI here. Similar code does not automatically mean duplication to me, if the logic os the same then it is. Since #my_organizations and #my_locations can potentially do different things I don't think it's duplication, only the name pattern is the same.
* metaprogramming is cool when you write it, pretty bad when you read it or try to grep source-code, I don't think the similarity here is justification to generate the code dynamically. The taxonomy code is already pretty hard to understand and big portion of this is because we use [this](https://github.com/theforeman/foreman/blob/develop/app/models/concerns/taxonomix.rb#L66)

The more of this we add the more we'll have to remove when we get to it. If someone later decides it's better to keep it this way, it's not too bad duplication since it's in one file near each other. I think worse duplication is that you can't spot easily (multiple methods, files etc).